core/templates/dev/head/components/question-directives/question-player/question-player.directive.ts

```diff
@@ -197,9 +198,68 @@ oppia.directive('questionPlayer', [
               ctrl.questionPlayerConfig.resultActionButtons.length > 0);
           };
 
+          var getWorstSkillId = function() {
+            var minScore = Number.MAX_VALUE;
+            var worstSkillId = '';
+            Object.keys(ctrl.scorePerSkillMapping).forEach(function(skillId) {
+              var skillScoreData = ctrl.scorePerSkillMapping[skillId];
+              var scorePercentage = skillScoreData.score / skillScoreData.total;
+              if (scorePercentage < minScore) {
+                minScore = scorePercentage;
+                worstSkillId = skillId;
+              }
+            });
+            return worstSkillId;
+          };
+
+          var openConceptCardModal = function(skillIds) {
+            var skills = [];
+            skillIds.forEach(function(skillId) {
+              skills.push(
+                ctrl.scorePerSkillMapping[skillId].description);
+            });
+            $uibModal.open({
+              templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
+                '/components/question-directives/question-player/' +
+                'review-and-retry-modal.template.html'
+              ),
+              backdrop: true,
+              controller: [
+                '$scope', '$uibModalInstance', '$window',
+                'UrlService',
+                function(
+                    $scope, $uibModalInstance, $window,
+                    UrlService) {
+                  $scope.failedSkillIds = skillIds;
+                  $scope.failedSkills = skills;
+                  $scope.index = 0;
+                  $scope.currentSkill = $scope.failedSkills[$scope.index];
+
+                  $scope.isLastConceptCard = function() {
+                    return $scope.index === $scope.failedSkills.length - 1;
+                  };
+
+                  $scope.closeModal = function() {
+                    $uibModalInstance.dismiss('cancel');
+                  };
+
+                  $scope.goToNextConceptCard = function() {
+                    $scope.index++;
+                    $scope.currentSkill = $scope.failedSkills[$scope.index];
+                  };
+
+                  $scope.retryTest = function() {
```

Comment:
Just to clarify - this 'Retry Test' button will also show up in the boost score modal right?

Reply:
Yes it will